From wfp5p@virginia.edu  Mon Jul 27 14:06:58 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 27 Jul 2009 16:47:27 -0400
Subject: Staging: hv: fix blkvsc_open() parameters
To: greg@kroah.com
Cc: hjanssen@microsoft.com
Message-ID: <1248727665-23969-6-git-send-email-wfp5p@virginia.edu>


blkvsc_open() had the wrong parameter list for  struct
block_device_operations

Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/blkvsc_drv.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/hv/blkvsc_drv.c
+++ b/drivers/staging/hv/blkvsc_drv.c
@@ -124,7 +124,7 @@ static int blkvsc_probe(struct device *d
 static int blkvsc_remove(struct device *device);
 static void blkvsc_shutdown(struct device *device);
 
-static int blkvsc_open(struct inode *inode, struct file *filep);
+static int blkvsc_open(struct block_device *bdev,  fmode_t mode);
 static int blkvsc_release(struct inode *inode, struct file *filep);
 static int blkvsc_media_changed(struct gendisk *gd);
 static int blkvsc_revalidate_disk(struct gendisk *gd);
@@ -1320,9 +1320,9 @@ static void blkvsc_request(struct reques
 	}
 }
 
-static int blkvsc_open(struct inode *inode, struct file *filep)
+static int blkvsc_open(struct block_device *bdev,  fmode_t mode)
 {
-	struct block_device_context *blkdev = inode->i_bdev->bd_disk->private_data;
+	struct block_device_context *blkdev = bdev->bd_disk->private_data;
 
 	DPRINT_DBG(BLKVSC_DRV, "- users %d disk %s\n", blkdev->users, blkdev->gd->disk_name);
 
@@ -1331,7 +1331,7 @@ static int blkvsc_open(struct inode *ino
 	if (!blkdev->users && blkdev->device_type == DVD_TYPE)
 	{
 		spin_unlock(&blkdev->lock);
-		check_disk_change(inode->i_bdev);
+		check_disk_change(bdev);
 		spin_lock(&blkdev->lock);
 	}
 
